lib/liquid/combinators/general.ex

```diff
@@ -202,6 +202,15 @@ defmodule Liquid.Combinators.General do
       string(@start_tag)
     ])
     |> reduce({List, :to_string, []})
+    |> traverse({Liquid.Combinators.General, :check_empty, []})
+  end
+
+  def check_empty(_rest, args, context, _line, _offset) do
```

Comment:
I think its more efficient to do the if-else als 2 different function headers, where you pattern match on `args` being `""`

Reply:
Done